### DISCUSSION TITLE
[LDAP] Allow existing users to log in with LDAP instead of mealie auth

### DISCUSSION THREAD

**Discussion by cmintey** [mealie-recipes/mealie#2126] [Feature Request]
**Before submitting this feature request I have**
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
With the latest changes to LDAP auth, users who do not already exist in Mealie are created on the fly, which is great. However, existing users in mealie cannot log in with LDAP (without manually changing their password to `LDAP` in the database).

**(Optional): Suggest A Solution**
I propose a solution similar to how the LDAP plugin works in [Jellyfin](https://jellyfin.org/). In this plugin, there is an option for each user on which authentication provider should be used, either "default" or "LDAP". With this in place, the existing mealie users could be switched over to using LDAP auth instead of having to manually change their password, or deleting their account and having LDAP recreate it. 

This could be accomplished by adding a new column to the user table `auth_type`, which defaults to `mealie` (or whatever). Then on login, the flow would be as follows:
1. find user
2. user exists: check `auth_type`. user not exist: proceed as normal
  `auth_type == "mealie"` --> login as normal
  `auth_type == "ldap"` --> login with ldap

![image](https://user-images.githubusercontent.com/35710697/218498369-92f063e3-8e5d-4c6c-9146-90c477719e6c.png)

**Additional Information**
- [x] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by cmintey** [comment]
Added auth type column
![image](https://user-images.githubusercontent.com/35710697/219500601-0204c541-04f7-45fd-b57b-aaf00f08e49e.png)

Allow selection in the user edit page: 
![image](https://user-images.githubusercontent.com/35710697/219500759-ea85b80c-d2d4-4267-8fba-c9592b1d3d4f.png)

**Reply by cmintey** [reply]
User creation
![image](https://user-images.githubusercontent.com/35710697/219502162-120d37f6-09f8-4744-994d-58f7cd215922.png)

**Reply by Zackptg5** [reply]
Would there be a way to set the default "auth_type" for general logins/new users? Would make the most sense for ldap to be default if one exists and mealie otherwise

**Reply by cmintey** [reply]
Mealie is the default when you create a user from this screen. If you want to create LDAP users, you don’t need to manually create an account because one will be created on a user first login (and the auth_type will be `LDAP`)

**Reply by Zackptg5** [reply]
Gotcha. Tested out latest nightly and works like a charm. I just create user in ldap, login to mealie with new user and auto-set to ldap for that user. Thanks for all your work on this, greatly appreciated :)
